pandas/_libs/lib.pyx

```diff
@@ -1864,8 +1889,8 @@ cdef class StringValidator(Validator):
         return issubclass(self.dtype.type, np.str_)
 
     cdef bint is_valid_null(self, object value) except -1:
-        # We deliberately exclude None / NaN here since StringArray uses NA
-        return value is C_NA
+        # Override to exclude float('Nan') and complex NaN
+        return value is None or value is C_NA or value is np.nan
```

Comment:
similar question to above, can `is` always be used with `np.nan` or should use np.isnan instead?

Reply:
I think it's a singleton(like None) so ``is`` should always work. I think @jreback mentioned above that np.isnan also returns True for float('nan') which is undesirable.

Reply:
In *theory* you can have NaNs that are not identical to np.nan (eg quiet vs signaling NaN):

```
In [78]: arr = np.frombuffer(b'\x01\x00\x00\x00\x00\x00\xf0\x7f\x00\x00\x00\x00\x00\x00\xf0?', dtype="float64")

In [79]: arr
Out[79]: array([nan,  1.])

In [80]: arr[0]
Out[80]: nan

In [81]: type(arr[0])
Out[81]: numpy.float64

In [82]: arr[0] is np.nan
Out[82]: False

In [83]: np.isnan(arr[0])
Out[83]: True
```

That's probably a case we don't really care about here (it would be very strange to get this in an object dtype array with strings), but nonetheless it's maybe good to be consistent internally how NaN is checked? We could have a version of `checknull` that doesn't do the NaT and decimal check?

Reply:
switched to using util.is_nan which should accept this and all other floating nans.